executor/partition_table.go

```diff
@@ -54,6 +59,9 @@ type nextPartitionForIndexLookUp struct {
 
 func (n nextPartitionForIndexLookUp) nextPartition(ctx context.Context, tbl table.PhysicalTable) (Executor, error) {
 	n.exec.table = tbl
+	if err := updateDAGRequestTableID(ctx, n.exec.dagPB, tbl.Meta().ID, tbl.GetPhysicalID()); err != nil {
```

Comment:
There's no need to update the dag for index lookup. Because it's impossible to build an index lookup with TiFlash.